Adds Azure public cloud reference doc

r7vme opened the pull request:
## Description

I've really struggled while trying to install k8s on Azure with Calico (own terraform), acs-engine reverse engineering took at least few days before i found proper Calico mode and Kubernetes configuration. So i thought that it can be useful for others to have explicit documentation on how Calico should be installed in Azure.

Describes only available installation option on Azure in policy-only
mode

Closes: https://github.com/projectcalico/calico/issues/220
Related-to: https://github.com/projectcalico/calicoctl/issues/949


## Todos

- [x] Documentation

## Release Note

<!-- Writing a release note:
- By default, no release note action is required.
- If you're unsure whether or not your PR needs a note, ask your reviewer for guidance.
- If this PR requires a release note, update the block below to include a concise note describing
  the change and any important impacts this PR may have.
-->

```release-note
None required
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/projectcalico/calico?pullRequest=1403) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/projectcalico/calico?pullRequest=1403) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/projectcalico/calico?pullRequest=1403) it.</sub>

r7vme commented:
I'm not sure how can i prevent this error
```
- /_site/master/reference/public-cloud/azure.html
  *  External link https://docs.projectcalico.org/latest/reference/public-cloud/azure failed: 404 No error
htmlproofer 3.3.1 | Error:  HTML-Proofer found 1 failure!
make: *** [htmlproofer] Error 1
```

fasaxc commented on master/getting-started/kubernetes/installation/azure.md:
```diff
@@ -15,5 +17,6 @@ If the out-of-the-box solutions listed above don't meet your requirements, you c
 on Azure using one of our [self-hosted manifests][self-hosted], or by [integrating {{site.prodname}} with your own configuration management][integration-guide].
 
 [acs-engine]: https://github.com/Azure/acs-engine/blob/master/docs/kubernetes.md
+[azure-reference]: {{site.baseurl}}/{{page.version}}/reference/public-cloud/azure
```
I think you just need to make this a relative URL to avoid the htmlproofer error.

r7vme replied:
Noop it does not help, i did `[azure-reference]: ../../../reference/public-cloud/azure`.

Problem in other part, generated `reference/public-cloud/azure.html` page has this 
```
<link rel="canonical" href="https://docs.projectcalico.org/latest/reference/public-cloud/azure" />
```

So as i understand it will be checking real upstream, which does not have this page.

r7vme replied:
Fixed for now by adding `no_canonical: true` to the page. Please take look on content itself.

r7vme commented:
@emanic can you please take look?

emanic commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+
+#### Enabling IP forwarding
+
+To allow pods traffic make sure VM network interfaces have [ip forwarding enabled][AzureIPForward]{:target="_blank"} in Azure.
```
"pods" --> "pod"
"ip" --> "IP"

emanic commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+
+#### Enabling Kubernetes pod CIDR assignment
+
+To enable automatic pod CIDR assignement make sure Kubernetes controller manager has allocate-node-cidrs set to true
```
"assignement" --> "assignment"
"allocate-node-cidrs" --> "`allocate-node-cidrs`" (backticks)

emanic commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+#### Enabling Kubernetes pod CIDR assignment
+
+To enable automatic pod CIDR assignement make sure Kubernetes controller manager has allocate-node-cidrs set to true
+and proper subnet in cluster-cidr parameter. Make sure that selected pods subnet is a part of your Azure virtual network ip range.
```
"that selected pods subnet" --> "that the selected pod's subnet"
"ip" --> "IP"

emanic commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+and proper subnet in cluster-cidr parameter. Make sure that selected pods subnet is a part of your Azure virtual network ip range.
+You also must have Kubernetes Azure cloud provider configured with your routing table name in configuration file.
+
+#### Why {{site.prodname}} networking is not supported in Azure?
```
"Why {{site.prodname}} networking is not supported in Azure?" --> "Why doesn't Azure support {{site.prodname}} networking?"

emanic commented:
@r7vme Sorry for the delay and thank you so much for your contribution! I've left a few comments, if you could potentially address these, that would help us move your PR forward. 

It would be great if you could port these changes to the `v3.0` directory, squash your commits, and rebase on top of the latest master. Please let me know if you have any difficulty with these or anything else.

Thanks again, we do appreciate your contribution!

r7vme commented:
@emanic thanks for review. I addressed inline comments and backported changes to `v3.0`. Please take a look.

emanic commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+---
+
+{{site.prodname}} in [Microsoft Azure][Azure]{:target="_blank"} supported in [policy-only][PolicyMode] mode.
+Networking part covered by Azure user-defined routes. {{site.prodname}} IPAM needs
```
Can you insert an "is" just before supported? That would make the sentence more correct.

This sentence "Networking part covered by Azure user-defined routes." is problematic grammatically. I think you are definitely missing an article and a verb here. However, since this sentence just before this one says almost the same thing and you cover this topic so well just below, I think you can just delete this sentence.

caseydavenport commented on master/reference/public-cloud/azure.md:
```diff
@@ -0,0 +1,37 @@
+
+Only available option for traffic routing in Azure with {{site.prodname}} is [Azure user-defined routes][AzureUDR]{:target="_blank"}.
+
+##### Azure user-defined routes
```
An alternative to this would be to use flannel networking (a la "canal") since Azure is totally fine with vxlan.

emanic commented:
LGTM. Thanks @r7vme for the updates. If possible, could you squash the commits? That will allow us to potentially merge this PR, ideally squashed to a single commit. 

If you can, after that, it might be nice if you could rebase on top of the latest master (`git rebase upstream/master` (you may need to update your local master branch first)). This will likely resolve the `htmlproofer` errors. I hope that's not too much trouble? If so, let me know.... Thanks again!

r7vme commented:
@emanic thanks.

- squashed to 1 commit
- rebased to latest master (did it yesterday, but for some reason CI was failing multiple times yesterday.)

emanic commented:
Great thanks @r7vme ! Yes, we have been having trouble with Semaphore for the last couple of days, that's causing various issues. Looks like it's passed now. Thanks again!
